src/main/java/com/maniek/medicalclinic/service/VisitService.java

```diff
@@ -0,0 +1,55 @@
+package com.maniek.medicalclinic.service;
+
+import com.maniek.medicalclinic.exception.PatientNotFoundException;
+import com.maniek.medicalclinic.exception.VisitIllegalArgumentException;
+import com.maniek.medicalclinic.exception.VisitNotFoundException;
+import com.maniek.medicalclinic.mapper.VisitMapper;
+import com.maniek.medicalclinic.model.dto.VisitDTO;
+import com.maniek.medicalclinic.model.entity.Patient;
+import com.maniek.medicalclinic.model.entity.Visit;
+import com.maniek.medicalclinic.repository.PatientRepository;
+import com.maniek.medicalclinic.repository.VisitRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class VisitService {
+    private final VisitRepository visitRepository;
+    private final VisitMapper visitMapper;
+    private final PatientRepository patientRepository;
+
+    public List<Visit> showVisit() {
+        return visitRepository.findAll();
+    }
+
+    public VisitDTO addVisit(VisitDTO visitDTO) {
+        if (visitRepository.findByTerm(visitDTO.getTerm()).isPresent()) {
+            throw new VisitIllegalArgumentException("Illegal visit ");
```

Comment:
Moze powinnismy zastanowic sie czy nie lepiej byloby dac message, ktory wiecej mowi tj. np w tym przypadku 
"Date is unavailable"